File: pinyougou-shop-web/src/main/webapp/js/service/goodsService.js
app.service('goodsService',function ($http) {

    //按条件查询分页列表
    this.queryAllGoodsListByConditionPage=function (current,size,searchEntity) {
        return $http.post("../goods/queryAllGoodsListByConditionPage.do?current=" + current + "&size=" + size, searchEntity);
    };

    this.selectByOne=function (id) {
        return $http.get("../goods/selectByOne.do?id="+id);
    };

   this.add=function (entity) {
       return $http.post("../goods/add.do",entity);
   };
    this.update=function (entity) {
        return $http.post("../goods/update.do",entity);
    };
    this.delete=function (ids) {
        return $http.post("../goods/delete.do?ids="+ids);
    };
    //提交审核
    this.audit=function (ids,status) {
        return $http.get("../goods/audit.do?ids="+ids+"&status="+status);
    }
    this.isMarketable=function (ids,status) {
        return $http.get("../goods/isMarketable.do?ids="+ids+"&status="+status);
    }
});

File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/service/GoodsService.java
package com.pinyougou.service;

import com.baomidou.mybatisplus.plugins.Page;
import com.baomidou.mybatisplus.service.IService;
import com.pinyougou.pojo.Goods;
import com.pinyougou.pojo.Item;
import com.pinyougou.vo.GoodsVo;

import java.util.List;

/**
 * <p>
 *  服务类
 * </p>
 *
 * @author ZhiCaiLi
 * @since 2018-10-02
 */
public interface GoodsService extends IService<Goods> {
    /**
     * 根据条件查询分页列表
     * @param page 分页条件
     * @param goods 查询条件
     * @return
     */
    public Page<Goods> queryAllGoodsListByConditionPage(Page<Goods> page, Goods goods);

    /**
     * 根据id获取实体
     * @param id
     * @return
     */
    public GoodsVo selectByOne(Long id);

    /**
     * 增加
     * @param goodsVo
     */
    public void add(GoodsVo goodsVo);

    /**
     * 更新
     * @param goodsVo
     */
    public  void update(GoodsVo goodsVo);

    /**
     * 根据商品id和状态查询item表的信息
     * @param ids
     * @param status
     * @return
     */
    public List<Item> selectItemListByGoodsIdandStatus(Long[] ids,String status);
}
